onnxruntime/core/framework/error_code.cc

```diff
@@ -12,9 +12,9 @@ struct OrtStatus {
   char msg[1];  // a null-terminated string
 };
 
-ORT_API(OrtStatus*, OrtCreateStatus, OrtErrorCode code, _In_ const char* msg) {
-  assert(!(code == 0 && msg != nullptr));
-  size_t clen = strlen(msg);
+ORT_API(OrtStatus*, OrtCreateStatus, OrtErrorCode code, _In_opt_ const char* msg) {
```

Comment:
What's the use case where we don't want to provide an error message?

Reply:
In the C++ API, common::Status doesn't require an error message.